website/docs/backends/types/consul.html.md

```diff
@@ -12,7 +12,7 @@ description: |-
 
 Stores the state in the [Consul](https://www.consul.io/) KV store at a given path.
 
-This backend supports [state locking](/docs/state/locking.html).
+This backend supports [state locking](/docs/state/locking.html). Note that state storage with this backend is subject to the [maximum 512KB limit](https://www.consul.io/docs/faq.html#q-what-is-the-per-key-value-size-limitation-for-consul-39-s-key-value-store-) in Vault. See also the `gzip` option below.
```

Comment:
Indeed. It's not set to true by default because it's convenient for data in Consul to be human-readable. But not setting it creates the risk that you'll get broken someday. (Even if you _do_ set it, your gzipped payload might still be too large, but still...)

I agree that there isn't really an absolute thing we can state here but perhaps we can give some more specific direction on those tradeoffs of human-readable stuff in Consul vs. reducing the risk of size overages.

"Consider using another backend entirely" ought to be in there _somewhere_, I think, but I'm not sure how best to include it in an actionable way.